script/prerender/manage.ts

```diff
@@ -0,0 +1,397 @@
+// tslint:disable:no-console
+
+/*
+  Manages a fleet of spot instances
+  Creates a cloudformation stack with a spot fleet and SQS queues, preloads all the book ToCs,
+  queues up each page in the queue to be prerendered, and deletes the stack at the end
+*/
+
+import './setup';
+
+import {
+  CloudFormationClient,
+  CreateStackCommand,
+  DeleteStackCommand,
+  DescribeStacksCommand,
+  waitUntilStackCreateComplete,
+} from '@aws-sdk/client-cloudformation';
+import {
+  GetQueueAttributesCommand,
+  ReceiveMessageCommand,
+  SendMessageBatchCommand,
+  SendMessageCommand,
+  SQSClient,
+} from '@aws-sdk/client-sqs';
+import omit from 'lodash/fp/omit';
+import path from 'path';
+import Loadable from 'react-loadable';
+import asyncPool from 'tiny-async-pool';
+import { makeUnifiedBookLoader } from '../../src/app/content/utils';
+import { assertDefined } from '../../src/app/utils';
+import config from '../../src/config';
+import BOOKS from '../../src/config.books';
+import createArchiveLoader from '../../src/gateways/createArchiveLoader';
+import createOSWebLoader from '../../src/gateways/createOSWebLoader';
+import { readFile } from '../../src/helpers/fileUtils';
+import { globalMinuteCounter, prepareBookPages } from './contentPages';
+import createRedirects from './createRedirects';
+import { writeAssetFile } from './fileUtils';
+
+const {
+  ARCHIVE_URL,
+  CODE_VERSION,
+  OS_WEB_URL,
+  REACT_APP_ARCHIVE_URL,
+  REACT_APP_OS_WEB_API_URL,
+  RELEASE_ID,
+} = config;
+
+// Increasing this too much can lead to connection issues and greater memory usage in the manager
+const MAX_CONCURRENT_BOOKS = 5;
+
+// The worker fleet is automatically terminated after this many seconds
+// This is insurance in case this process gets stuck or crashes without deleting the workers stack
+const PRERENDER_TIMEOUT_SECONDS = 1800;
+
+// Abort the build if the workers stack is not ready after this many seconds
+const WORKERS_DEPLOY_TIMEOUT_SECONDS = 120;
+
+const BUCKET_NAME = process.env.BUCKET_NAME || 'sandbox-unified-web-primary';
+const BUCKET_REGION = process.env.BUCKET_REGION || 'us-east-1';
+const PUBLIC_URL = process.env.PUBLIC_URL || `/rex/releases/${RELEASE_ID}`;
+const RELEASE_ID_SUFFIX = RELEASE_ID.split('/', 2)[1];
```

Comment:
this doesn't really have any value when split out, i think it would make more sense to string replace the blocked characters rather than just using the last bit

Reply:
Done